presto-main/src/main/java/io/prestosql/operator/scalar/MathFunctions.java

```diff
@@ -690,35 +690,51 @@ public static long round(@SqlType(StandardTypes.BIGINT) long num)
     @SqlType(StandardTypes.TINYINT)
     public static long roundTinyint(@SqlType(StandardTypes.TINYINT) long num, @SqlType(StandardTypes.INTEGER) long decimals)
     {
-        // TODO implement support for `decimals < 0`
-        return num;
+        return roundLong(num, decimals);
     }
 
     @Description("round to nearest integer")
     @ScalarFunction("round")
     @SqlType(StandardTypes.SMALLINT)
     public static long roundSmallint(@SqlType(StandardTypes.SMALLINT) long num, @SqlType(StandardTypes.INTEGER) long decimals)
     {
-        // TODO implement support for `decimals < 0`
-        return num;
+        return roundLong(num, decimals);
     }
 
     @Description("round to nearest integer")
     @ScalarFunction("round")
     @SqlType(StandardTypes.INTEGER)
     public static long roundInteger(@SqlType(StandardTypes.INTEGER) long num, @SqlType(StandardTypes.INTEGER) long decimals)
     {
-        // TODO implement support for `decimals < 0`
-        return num;
+        return roundLong(num, decimals);
     }
 
     @Description("round to nearest integer")
     @ScalarFunction
     @SqlType(StandardTypes.BIGINT)
     public static long round(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.INTEGER) long decimals)
     {
-        // TODO implement support for `decimals < 0`
-        return num;
+        return roundLong(num, decimals);
+    }
+
+    private static long roundLong(long num, long decimals)
+    {
+        if (decimals >= 0) {
+            return num;
+        }
+        else {
```

Comment:
Can we remove this `else` to avoid unnecessary nest?
```java
        if (decimals >= 0) {
            return num;
        }

        double factor = Math.pow(10, -decimals);
...
```